From f5c7dff47b9ca8649b97763823d21a222f0d4fe1 Mon Sep 17 00:00:00 2001
From: "Maneendra,Pokuri Naga" <pnmaneendra@maxlinear.com>
Date: Thu, 24 Oct 2024 18:10:51 +0530
Subject: [PATCH] WLANRTSYS-85692 Recover mld_id of BSS in SIGHUP flow

RCA:
During 2 MultiLink VAP configurations, mld_id of the 1st ML VAP setting to INVALID value after SIGHUP
and also this param is not read from conf file we are facing issues while removing the MLD

Fix:
During every config reload we will recover the mld_id of each BSS based on Sibling and Vice Versa.

Fixup to WLANRTSYS-60843 MLO changes: AP MLD configuration.
---
 src/ap/hostapd.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/ap/hostapd.c b/src/ap/hostapd.c
index 16dbd16e3..a640d3f18 100644
--- a/src/ap/hostapd.c
+++ b/src/ap/hostapd.c
@@ -363,6 +363,10 @@ int hostapd_reload_config(struct hostapd_iface *iface)
 			hostapd_get_oper_centr_freq_seg1_idx(oldconf));
 		hapd->conf = newconf->bss[j];
 		hostapd_reload_bss(hapd);
+#ifdef CONFIG_IEEE80211BE_MXL_MLO
+		if (hapd->conf->mxl_bss_conf.mlo_enable && hapd->mxl_data.sibling_hapd)
+			hapd->conf->mld_id = hapd->mxl_data.sibling_hapd->conf->mld_id;
+#endif /* CONFIG_IEEE80211BE_MXL_MLO */
 	}
 
 	hostapd_config_free(oldconf);
-- 
2.43.0

